src/main/java/com/atlassian/jira/cloud/jenkins/common/client/JenkinsAppApi.java

```diff
@@ -153,7 +162,9 @@ protected String wrapInJwt(
             final JenkinsAppRequest request, final String secret, final Date expiryDate)
             throws JsonProcessingException {
         final String body = objectMapper.writeValueAsString(request);
-        logger.info(String.format("wrapping request in JWT: %s", body));
+        if (pipelineLogger != null) {
+            pipelineLogger.info(String.format("sending request to Jenkins app in Jira: %s", body));
+        }
```

Comment:
PipelineLogger has to be optional, because we don't always have access to a pipeline `PrintStream`. This is the case for the `PingApi` class, for example, which is triggered from the Jenkins config UI to check if the webtrigger URL is correct ... there is no Jenkins Pipeline in that context.

Reply:
In that case, should we do `this.pipelineLogger = PipelineLogger.noopInstance()` in the constructor? 
I am worry someone may forgot to check null some where.

Reply:
Good call. Will make the chance.